File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int cnt =0;
    int res = 0;
    public int kthSmallest(TreeNode root, int k) {
        inorderTraversal(root, k);
        return res;
    }
    public void inorderTraversal(TreeNode root, int k){
        if(root == null)return;
        inorderTraversal(root.left, k);
        cnt++;
        if(cnt == k){
            res = root.val;
            return;
        }
        inorderTraversal(root.right, k);
    }
}

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        minheap = []
        q = deque([root])
        while q:
            lq = len(q)
            for _ in range(lq):
                node = q.popleft()
                heapq.heappush(minheap, node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
        for _ in range(k-1):
            heapq.heappop(minheap)
        return heapq.heappop(minheap)

        